supabase/functions/_shared/helpers.ts

```diff
@@ -1,20 +1,46 @@
-import { corsHeaders } from "./cors.ts";
+import HandlebarsJS from 'https://esm.sh/handlebars';
+import Mustache from 'https://esm.sh/mustache';
+import { corsHeaders } from './cors.ts';
 
 export const returnPostgresError = (error: any) => {
-  return new Response(JSON.stringify({ error }), {
-    headers: {
-      ...corsHeaders,
-      "Content-Type": "application/json",
-    },
-    status: 400,
-  });
+    return new Response(JSON.stringify({ error }), {
+        headers: {
+            ...corsHeaders,
+            'Content-Type': 'application/json',
+        },
+        status: 400,
+    });
 };
 
 export const handlebarsHelpers = {
-  urlencode: function(s: string) {
-    return encodeURIComponent(s)
-  },
-  basicauth: function(user: string, password: string) {
-    return btoa(`${user}:${password}`);
-  }
-}
+    urlencode: function (s: string) {
+        return encodeURIComponent(s);
+    },
+    basicauth: function (user: string, password: string) {
+        return btoa(`${user}:${password}`);
+    },
+};
+
+export const compileTemplate = (template: string, data: any, connector_id: string) => {
+    (HandlebarsJS as any).registerHelper(handlebarsHelpers);
+
+    const compiledTemplate = (HandlebarsJS as any).compile(template);
+    const handlebarsOutput = compiledTemplate(data);
+
+    try {
+        //Duplicate for testing Mustache
+        const mustacheOutput = Mustache.render(template, {
+            ...data,
+            ...handlebarsHelpers,
+        });
+
+        if (handlebarsOutput !== mustacheOutput) {
+            console.log(`oauth:templates:mismatch:${connector_id}`, { template });
+        }
+    } catch (e) {
```

Comment:
Just want to double check whether this could cause an error to be swallowed. Would our handler still return an error if the template fails to compile? Seems like we'd want it to return a 500 in that case.

Reply:
My understanding is that this will swallow compilation error for the mustache template, but not Handlebars. We want to use Handlebars as the default and Mustache for comparison only at this stage, once we reach a stage where Handlebars and Mustache are consistently giving back the same rendering for all templates, then we will remove Handlebars and use Mustache as the main, removing the `try catch`